Создано по одному тесту на каждый эндпоинт контроллеров StudentContro…

Manikgog opened the pull request:
…ller и FacultyController, используя TestRestTemplate.

Manikgog committed:
Создано по одному тесту на каждый эндпоинт контроллеров StudentController и FacultyController, используя TestRestTemplate.

dmitry-bizin commented on src/test/java/ru/hogwarts/school/FacultyControllerTestRestTemplate.java:
```diff
@@ -0,0 +1,290 @@
+    @LocalServerPort
+    private int port;
+    @Autowired
+    private StudentController studentController;
```
Зачем контроллер здесь? он не нужен

dmitry-bizin commented on src/test/java/ru/hogwarts/school/FacultyControllerTestRestTemplate.java:
```diff
@@ -0,0 +1,290 @@
+    @Autowired
+    private StudentRepository studentRepository;
+    @Autowired
+    private FacultyController facultyController;
```
Зачем контроллер здесь? он не нужен

dmitry-bizin commented on src/test/java/ru/hogwarts/school/FacultyControllerTestRestTemplate.java:
```diff
@@ -0,0 +1,290 @@
+    @Autowired
+    private FacultyRepository facultyRepository;
+    @Autowired
+    private AvatarController avatarController;
```
Зачем контроллер здесь? он не нужен

dmitry-bizin commented on src/test/java/ru/hogwarts/school/FacultyControllerTestRestTemplate.java:
```diff
@@ -0,0 +1,290 @@
+    @Autowired
+    private AvatarController avatarController;
+    @Autowired
+    private AvatarService avatarService;
```
также сервисы лишние есть и репозитории, зачем? зачем лишние поля неиспользуемые?

dmitry-bizin commented on src/test/java/ru/hogwarts/school/FacultyControllerTestRestTemplate.java:
```diff
@@ -0,0 +1,290 @@
+        Assertions.assertThat(this
+                        .restTemplate
+                        .postForObject("http://localhost:" + port + "/faculties", FACULTY_1, Faculty.class))
+                .matches(faculty -> faculty.getName().equals(FACULTY_1.getName()) &&
+                        faculty.getColor().equals(FACULTY_1.getColor()), "данные факультетов равны");
```
почему бы здесь не вызвать метод test, который ты выше написал?

dmitry-bizin commented on src/test/java/ru/hogwarts/school/FacultyControllerTestRestTemplate.java:
```diff
@@ -0,0 +1,290 @@
+                .matches(faculty -> faculty.getName().equals(FACULTY_1.getName()) &&
+                        faculty.getColor().equals(FACULTY_1.getColor()), "данные факультетов равны");
+
+        Faculty fac = facultyRepository.findAll().stream().filter(faculty ->    // получение загруженного в базу данных объекта для его последующего удаления
+                        faculty.getName().equals(FACULTY_1.getName()) &&
+                        faculty.getColor().equals(FACULTY_1.getColor())
+        ).findFirst().get();
```
а почему бы не по id сразу достать? зачем получать всех из базы и потом искать в коде?

dmitry-bizin replied:
почему бы здесь не вызвать метод test, который ты выше написал?

dmitry-bizin commented on src/test/java/ru/hogwarts/school/FacultyControllerTestRestTemplate.java:
```diff
@@ -0,0 +1,290 @@
+        facultyRepository.save(FACULTY_1);
+
+        Faculty faculty = facultyRepository.findAll().stream().filter(f ->
+                        f.getName().equals(FACULTY_1.getName()) &&
+                        f.getColor().equals(FACULTY_1.getColor())
```
почему бы здесь не вызвать метод test, который ты выше написал?

dmitry-bizin commented on src/test/java/ru/hogwarts/school/FacultyControllerTestRestTemplate.java:
```diff
@@ -0,0 +1,290 @@
+        facultyRepository.save(FACULTY_1);
+
+        Faculty faculty = facultyRepository.findAll().stream().filter(f ->
+                        f.getName().equals(FACULTY_1.getName()) &&
+                        f.getColor().equals(FACULTY_1.getColor())
```
почему бы здесь не вызвать метод test, который ты выше написал?

dmitry-bizin commented on src/test/java/ru/hogwarts/school/FacultyControllerTestRestTemplate.java:
```diff
@@ -0,0 +1,290 @@
+                        .restTemplate
+                        .getForObject("http://localhost:" + port + "/faculties/" + FACULTY_1.getId(), Faculty.class))
+                .matches(f -> f.getId().equals(FACULTY_1.getId()) &&
+                        f.getName().equals(FACULTY_1.getName()) &&
+                        f.getColor().equals(FACULTY_1.getColor()), "данные факультетов равны");
```
почему бы здесь не вызвать метод test, который ты выше написал?

dmitry-bizin commented on src/test/java/ru/hogwarts/school/FacultyControllerTestRestTemplate.java:
```diff
@@ -0,0 +1,290 @@
+                                    .getBody())
+                            .stream()
+                            .toList();
+                    return actualList.size() == redList.size();
```
содержимое списков тоже хорошо бы проверить

dmitry-bizin commented on src/test/java/ru/hogwarts/school/FacultyControllerTestRestTemplate.java:
```diff
@@ -0,0 +1,290 @@
+                                    .getBody())
+                            .stream()
+                            .toList();
+                    return actualList.size() == redList.size();
```
содержимое списков тоже хорошо бы проверить

dmitry-bizin commented on src/test/java/ru/hogwarts/school/FacultyControllerTestRestTemplate.java:
```diff
@@ -0,0 +1,290 @@
+                                    .getBody())
+                            .stream()
+                            .toList();
+                    return actualList.size() == faculty_1_Students.size();
```
содержимое списков тоже хорошо бы проверить

dmitry-bizin commented on src/test/java/ru/hogwarts/school/FacultyControllerTestRestTemplate.java:
```diff
@@ -0,0 +1,290 @@
+     * Метод для очистки базы данных после тестирования
+     */
+    @AfterEach
+    public void clearDB(){
```
вот же у тебя есть метод очистки, зачем ты тогда в каждом тесте руками ещё удаляешь?

dmitry-bizin commented on src/test/java/ru/hogwarts/school/StudentControllerTestRestTemplate.java:
```diff
@@ -0,0 +1,286 @@
+	@LocalServerPort
+	private int port;
+	@Autowired
+	private StudentController studentController;
```
здесь тоже лишние бины инжектятся, которые не используются

dmitry-bizin commented on src/test/java/ru/hogwarts/school/StudentControllerTestRestTemplate.java:
```diff
@@ -0,0 +1,286 @@
+    /**
+     * Метод для очистки базы данных после тестирования
+     */
+	@AfterEach
```
тоже самое - зачем удалять в конце каждого метода, если есть after each метод с удалением всего

dmitry-bizin commented on src/test/java/ru/hogwarts/school/StudentControllerTestRestTemplate.java:
```diff
@@ -0,0 +1,286 @@
+                                    .getBody())
+                            .stream()
+                            .toList();
+                    return actualList.size() == list.size();
```
содержимое списков хорошо бы тоже сравнить

dmitry-bizin commented on src/test/java/ru/hogwarts/school/StudentControllerTestRestTemplate.java:
```diff
@@ -0,0 +1,286 @@
+                                    .getBody())
+                            .stream()
+                            .toList();
+                    return actualList.size() == list.size();
```
содержимое списков хорошо бы тоже сравнить

Manikgog committed:
В классах StudentControllerTestRestTemplate и FacultyControllerTestRestTemplate удалены лишние поля которые не используюстя при тестировании.
В классах FacultyControllerTestRestTemplate и StudentControllerTestRestTemplate из методов удалены строчки с вызовом метода для удаления записей из базы данных после тестирования в методе. После каждого метода вызывается метод для очистки БД - clearDB.
